BGAnimations/ScreenGameplay overlay/WhoIsCurrentlyWinning.lua

```diff
@@ -21,13 +27,29 @@ return Def.Actor{
 		p1_score = underlay:GetChild("P1Score")
 		p2_score = underlay:GetChild("P2Score")
 	end,
-	JudgmentMessageCommand=function(self) self:queuecommand("Winning") end,
+	JudgmentMessageCommand=function(self, params)
+		if not IsEX then
+			-- calculate the percentage DP manually rather than use GetPercentDancePoints.
+			-- That function rounds to the nearest .01%, which is inaccurate on long songs.
+			if params.Player == PLAYER_1 then
+				p1_dp = p1_pss:GetActualDancePoints() / p1_pss:GetPossibleDancePoints()
+			elseif params.Player == PLAYER_2 then
+				p2_dp = p2_pss:GetActualDancePoints() / p2_pss:GetPossibleDancePoints()
+			end
+			self:queuecommand("Winning")
+		end
+	end,
+	ExCountsChangedMessageCommand=function(self, params)
+		if IsEX then
+			if params.Player == PLAYER_1 then
+				p1_dp = params.ExScore
+			elseif params.Player == PLAYER_2 then
+				p2_dp = params.ExScore
+			end
+			self:queuecommand("Winning")
+		end
+	end,
```

Comment:
Maybe don't even add ExCountsChangedMessageCommand in ITG mode and vice versa.

Reply:
Both of them are relevant in both modes:

You can play ITG mode with EX scoring enabled or disabled
You can play FA+ mode with EX scoring enabled or disabled.